packages/playwright-core/src/server/dom.ts

```diff
@@ -689,28 +691,48 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
         this._page._timeoutSettings.timeout(options));
   }
 
-  async querySelector(selector: string, options: types.StrictOptions): Promise<ElementHandle | null> {
-    return this._page.selectors.query(this._context.frame, selector, options, this);
+  async querySelector(metadata: CallMetadata, selector: string, options: types.StrictOptions): Promise<ElementHandle | null> {
+    const controller = new ProgressController(metadata, this);
+    return controller.run(async progress => {
+      const { frame, info } = await this._frame.resolveFrameForSelector(progress, selector, options, this);
+      // If we end up in the same frame => use the scope again, live above was noop.
+      return this._page.selectors.query(frame, info, this._frame === frame ? this : undefined);
+    });
```

Comment:
Pass a timeout from TimeoutSettings, otherwise it is going to be zero. Same for other places.

Reply:
That is intended for element handle...